### PR TITLE
Lazy load python lab in LabViewsRenderer

### PR DESCRIPTION
**Pull request by molly-moen**
We saw an issue a while ago where pyodide in Python Lab was causing crashes in Music Lab on old iOS versions. This is because the assets for Python Lab were being loaded for every Lab2 lab due us importing all lab views in `LabViewsRenderer`. In order to re-enable pyodide I wrapped `PythonlabView` in [lazy](https://react.dev/reference/react/lazy) so it will only be loaded when we actually view Python Lab. We are ok with python lab potentially not working on old iOS versions since it is a prototype (I also haven't been able to repro this error anymore so it may be resolved). 

Getting lazy loading to work with a TypeScript component and our webpack config was a little tricky. I had to re-export the component in an `index.js` file in order to get things to work properly. The syntax is a little funky, but not too long. 

Using `lazy` requires wrapping the lazily-loaded component in a [Suspense](https://react.dev/reference/react/Suspense) component with a fallback. I refactored our loading spinner that fades in after 3 seconds and shows a slow loading message after 11 seconds into a separate component so it could be reused.

## Links
- jira ticket: [CT-299](https://codedotorg.atlassian.net/browse/CT-299)

## Testing story
Tested locally that Lab2 labs still all load as expected, and only Python Lab loads the pyodide-specific assets. 

## Follow-up work
If we like this, we can re-use this pattern for the other Lab2 views. I think it would be best to isolate each lab's assets so we can feel comfortable prototyping new labs without causing bugs on existing labs.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/views/Lab2Wrapper.tsx [hunk 1]
```diff
@@ -12,18 +12,15 @@ import classNames from 'classnames';
...
 
 import {ErrorFallbackPage, ErrorUI} from './ErrorFallbackPage';
 import Lab2Registry from '../Lab2Registry';
+import Loading from './Loading';
 
 export interface Lab2WrapperProps {
   children: React.ReactNode;
 }
 
-const noFade = window.location.href.includes('lab2-no-fade');
```
Comment (molly-moen) [inline apps/src/lab2/views/Lab2Wrapper.tsx:25]: @breville @sanchitmalhotra126 this didn't seem likely to be useful anymore so I removed it. Let me know if I should keep it.

Comment (breville) [reply apps/src/lab2/views/Lab2Wrapper.tsx:25]: This was a recent addition in https://github.com/code-dot-org/code-dot-org/pull/56732 and is functionality we still want.

Comment (molly-moen) [reply apps/src/lab2/views/Lab2Wrapper.tsx:25]: whoops will add back!

File: apps/src/lab2/views/LabViewsRenderer.tsx [hunk 2]
```diff
@@ -67,7 +82,14 @@ const appsProperties: {[appName in AppName]?: AppProperties} = {
   },
   pythonlab: {
     backgroundMode: false,
-    node: <PythonlabView />,
+    node: <div />,
+    lazyNode: lazy(() =>
+      import(
+        /* webpackChunkName: "pythonlab" */ '../../pythonlab/index.js'
```
Comment (sanchitmalhotra126) [inline apps/src/lab2/views/LabViewsRenderer.tsx:88]: Out of curiosity - does this actually create a separate webpack chunk, or is this still technically part of the lab2.js chunk?

Comment (molly-moen) [reply apps/src/lab2/views/LabViewsRenderer.tsx:88]: My understanding is yes. This comment actually does some magic so we get an asset request for `pythonlab.js` over the network (without it the request is for a file with a number as a name like `215.js`).

File: apps/src/lab2/views/LabViewsRenderer.tsx [hunk 3]
```diff
@@ -9,16 +9,24 @@ import {setUpBlocklyForMusicLab} from '@cdo/apps/music/blockly/setup';
 import MusicView from '@cdo/apps/music/views/MusicView';
 import StandaloneVideo from '@cdo/apps/standaloneVideo/StandaloneVideo';
 import classNames from 'classnames';
-import React, {useContext, useEffect, useState} from 'react';
+import React, {
+  ComponentType,
+  LazyExoticComponent,
```
Comment (ebeastlake) [inline apps/src/lab2/views/LabViewsRenderer.tsx:14]: What a name 😂

**Review by ebeastlake — APPROVED** [review]

**Pull request merged**